### PR TITLE
Implement auth + Cleanup main()

### PR DESCRIPTION
**Pull request by adamvalt**

### CODE REVIEW

**Comment by adamvalt** [comment]
mergnem az po #61, bude tam dost konfliktov

File: src/handlers/dashboard.rs [hunk 1]
```diff
@@ -9,18 +11,14 @@ use axum::{
     Extension,
 };
 
-use crate::common::repository::DbReadAll;
```
Comment (KovacFilip) [inline src/handlers/dashboard.rs:12]: Why delete this line? We use the ReadAll in game_repository.read_all() - for that I think we need that imported here.

File: src/handlers/dashboard.rs [hunk 2]
```diff
@@ -9,18 +11,14 @@ use axum::{
...
-
+/// # Panics
```
Comment (KovacFilip) [inline src/handlers/dashboard.rs:14]: When does it panic? Isn't it because the DbReadAll isn't imported?

**Review by KovacFilip — APPROVED** [review]

File: src/handlers/dashboard.rs (hunk 1 above)
Comment (adamvalt) [reply src/handlers/dashboard.rs:12]: It has simply moved to line 3.

File: src/handlers/dashboard.rs (hunk 2 above)
Comment (adamvalt) [reply src/handlers/dashboard.rs:14]: Probably related to the unhandled unwraps

Comment (adamvalt) [reply src/handlers/dashboard.rs:14]: The comment was auto added by clippy

**Pull request merged**